packages/async-iterator/lib/async-iterator.js

```diff
@@ -0,0 +1,68 @@
+const DefaultBindings = require('@serialport/bindings')
+const debug = require('debug')('serialport/async-iterator')
+/**
+ * A transform stream that does something pretty cool.
+ * @param {Object} options open options
+ * @example ```
+// To use the `AsyncIterator` interface:
+const { open, list } = require('@serialport/async-iterator')
+const ports = await list()
+const arduinoPort = ports.find(info => (info.manufacture || '').includes('Arduino'))
+const port = await open(arduinoPort)
+
+// read bytes until close
+for await (const bytes of port) {
+  console.log(`read ${bytes.length} bytes`)
+}
+
+// read 12 bytes
+const { value, end } = await port.next(12)
```

Comment:
Go with `port.read(byteCount)` as it'll make the interface more familiar for those who know WHATWG Streams

Reply:
Oh, it looks like I may be wrong about read(n) vs next()

Reply:
So yeah, it should be next() but I don't think it takes any arguments: 

https://jakearchibald.com/2017/async-iterators-and-generators/